halo2_proofs/tests/frontend_backend_split.rs

```diff
@@ -42,7 +42,7 @@ struct MyCircuitConfig {
 
     // Copy constraints between columns (a, b) and (a, d)
 
-    // A dynamic lookup: s_lookup * [1, a[0], b[0]] in s_ltable * [1, d[0], c[0]]
+    // A dynamic lookup: s_lookup * [a[0], b[0]] in s_ltable * [d[0], c[0]]
```

Comment:
Sorry but I think my suggestion here https://github.com/privacy-scaling-explorations/halo2/pull/347#discussion_r1629582280 is not complete.

Take a look at this example based on the updated lookup expressions of this circuit:
| offset | s_lookup | a   | b   | s_ltable | c   | d   |
| ------ | -------- | --- | --- | -------- | --- | --- |
|  0     |  1       |  0  |  0  |    1     | 2   | 4   |
|  1     |  1       |  2  |  4  |    1     | 3   | 9   |
|  2     |  0       |  0  |  0  |    0     | 0   | 0   |

At offset 0 we have input `1 * [0, 0] = [0, 0]` which matches the table at offset 2 `0 * [0, 0] = [0, 0]` but that's undesirable, because we are matching a table entry that has `s_ltable = 0`.

So checking that at least one table expression uses a selector/fixed is not enough.  Maybe the correct check is that at least one table expression **is** a selector/fixed.

Reply:
Thanks for catch! @ed255 
So, I think we reached a point that the `lookup_any` api should include the `table expression` which **is** a selector/fixed.  
Otherwise, it can be used for soundness error by malicious prover.
Also, from #335, we can see that when the single fixed col is used for `lookup_any` api, we should recommend user to use the `lookup` api.
Then, the summary is that we don't have any way to prevent the abuse of `lookup_any` api use, and should remove `lookup_any` api. right?